### ISSUE TITLE
Don't update to beta releases

### ISSUE THREAD

**Issue by frostney** [peerigon/updtr#7]
I've noticed that `updtr` will update a dependency to its beta release if one is available. (I assume the same holds true for release candidates, although I haven't tested that specifically.) I feel this is usually not the expected behavior. Is it possible to make this behavior available under a flag and only use stable releases by default?

**Comment by matthaias** [comment]
Thanks for your report. But i think i need more information. Which dependency was updated to a beta release? I just tried it with webpack as a dependency, there also exists a beta (`2.0.1-beta`), but updtr installed me the last stable version (`1.12.9`).

```
➜  updtr-test git:(master) ✗ updtr
Found 2 outdated modules

Starting to update your modules...
1/2 	 ● istanbul 0.4.1 success
2/2 	 ● webpack 1.12.9 success

Finished
```

`npm show webpack` shows me informations about `dist-tags`: `{ latest: '1.12.9', beta: '2.0.1-beta' }`. 

According to the npm documentation about `dist-tags` the latest published version is the `latest`-tag, if no tag is set:

> Publishing a package sets the "latest" tag to the published version unless the --tag option is used. For example, npm publish --tag=beta.

updtr installs the `latest` version. Maybe `latest` is in your package the beta tag and this is the reason why it was installed? 

So please let me know the package to find out the reason :)

**Comment by frostney** [comment]
I've run into this with the package `babel-eslint` (https://www.npmjs.com/package/babel-eslint) and you're absolutely right, the beta version is marked as its latest release.

It's probably more of a feature request than a bug: I guess what I'm asking for is an additional check for when a pre-release is found under the `latest` tag to rather install the version that is not a pre-release.

I have to admit, I've made the mistake of publishing a pre-release version as the latest version quite a few times myself. :)

**Comment by matthaias** [comment]
Thanks for the information. But i think it is a good idea to check if it is a unstable version and search the last stable version just to be sure. I think i can check this tomorrow.

**Comment by matthaias** [comment]
After thinking about that again, i am not sure anymore if this problem should be solved by updtr and just install the last stable version. Maybe updtr should just ignore modules with a *beta* release tagged under *latest*. What do you think about this?

**Comment by frostney** [comment]
:+1: I think this is a good solution

**Comment by matthaias** [comment]
I just pushed a first shot to branch `feature/ignore-unstable-versions`. Can you test it please?

**Comment by frostney** [comment]
It's still trying to install a beta version for me :(

<img width="431" alt="screen shot 2015-12-02 at 22 22 12" src="https://cloud.githubusercontent.com/assets/554648/11544677/66485e98-9943-11e5-9f2d-00aa679d3075.png">

**Comment by matthaias** [comment]
Ok, that was a fail... now it should work:

```
➜ npm i babel-eslint@4.1.6 --save
├── babel-eslint@4.1.6


➜ npm i servus.js@0.0.1 --save
└── servus.js@0.0.1

➜ updtr
Found 1 outdated module

Starting to update your modules...
1/1 	 ● servus.js 0.0.2 success

Finished
```

It just updated `servus.js` and not `babel-eslint` anymore. The check is for `alpha`, `beta` and `rc` for the moment. Do you know any other strings that mark a version as "unstable"?

**Comment by frostney** [comment]
Thanks for fixing this so quickly.

As for unstable versions in general, according to the [SemVer](http://semver.org/) specification, any version number with a hyphen is considered a pre-release ("unstable") version.

> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphen [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92.

**Comment by matthaias** [comment]
No worries. I will change it to versions with hyphen and merge it then. Thanks for testing and the information.

**Comment by matthaias** [comment]
Thanks for your pull request. Published with `v0.1.5`.
